packages/ember-auto-import/ts/webpack.ts

```diff
@@ -129,18 +135,59 @@ export default class WebpackBundler extends Plugin implements Bundler {
     //
     // cast is ok because we passed needsCache to super
     let stagingDir = realpathSync(this.cachePath!);
+    let outputTargets = [
+      {
+        path: join(this.outputPath, 'assets'),
+        bundleNames: [
+          ...this.opts.bundles.names,
+          ...this.opts.bundles.lazyEngines,
+        ],
+      },
+    ];
+
+    let webpackBundleConfigs: Configuration[] = [];
+
+    outputTargets.forEach(({ path, bundleNames }) => {
+      let entry: { [name: string]: { import: string[]; dependOn?: string[] } } =
+        {};
+      bundleNames.forEach((bundle) => {
+        entry[bundle] = {
+          import: [join(stagingDir, 'l.js'), join(stagingDir, `${bundle}.js`)],
+        };
 
-    let entry: { [name: string]: string[] } = {};
-    this.opts.bundles.names.forEach((bundle) => {
-      entry[bundle] = [
-        join(stagingDir, 'l.js'),
-        join(stagingDir, `${bundle}.js`),
-      ];
+        if (bundle !== 'app') {
+          // TODO: should this apply to custom entry props?
+          entry[bundle].dependOn = ['app'];
```

Comment:
I think a lazy engine can depend on a lazy engine too. The dependency graph is more complicated than just app vs flat set of lazy engines.